### PR TITLE
4677 sse support for wasm32

### PR DESCRIPTION
**Pull request by FirelightFlagboy**

### CODE REVIEW

File: libparsec/crates/platform_sse/src/backoff.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::time::Duration;
+
+pub struct Limited {
```
Comment (touilleMan) [inline libparsec/crates/platform_sse/src/backoff.rs:5]: We should also provide some test for this (this should be pretty trivial)

**Comment by FirelightFlagboy** [comment]
Superseded by #5084

File: libparsec/crates/platform_sse/src/backoff.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    fn get_duration_to_wait(&self) -> Duration {
+        let factor = match self.attempt {
+            0 => 0,
+            1 => 1,
+            2 => 5,
+            3 => 10,
+            4 => 30,
+            _ => 60,
+        };
+        self.duration.mul_f64(f64::from(factor))
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:44]: I'm missing something ? $min(30, 45) = 30$

[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:44]: 3e876be34

File: libparsec/crates/platform_sse/src/backoff.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            attempt: 0,
+            duration: Duration::from_secs(1),
+        }
+    }
+
+    pub fn reset(&mut self) {
+        self.attempt = 0;
+    }
+
+    pub fn set_retry(&mut self, duration: Duration) {
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:23]: ccb123f95

File: libparsec/crates/client_connection/tests/cmds.rs [hunk 4]
```diff
@@ -91,18 +90,19 @@ async fn authenticated_sse(env: &TestbedEnv) {
         }
     };
 
-    // Request sent before sse started, will be ignored
+    log::trace!("Request sent before sse started, will be ignored");
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/client_connection/tests/cmds.rs:93]: In #5084 we don't use log in test, closing the discussion

File: libparsec/crates/platform_sse/src/backoff.rs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+impl Limited {
+    pub async fn wait(&mut self) {
+        let duration_to_wait = self.get_duration_to_wait();
+        self.attempt += 1;
+        libparsec_platform_async::sleep(duration_to_wait).await;
+    }
+
+    fn get_duration_to_wait(&self) -> Duration {
+        let factor = match self.attempt {
+            0 => 0,
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:37]: With [3e876be34](https://github.com/Scille/parsec-cloud/commit/3e876be34) this discussion is outdated

File: libparsec/crates/platform_sse/src/backoff.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:5]: e67cfcb74

Comment (FirelightFlagboy) [reply libparsec/crates/platform_sse/src/backoff.rs:5]: bb09fac84

**Pull request closed without merging**